### DISCUSSION TITLE
How to exclude the lookahead part from colouring

### DISCUSSION THREAD

**Discussion by datatraveller1** [BurntSushi/ripgrep#2328] [General] [answered]
The input file "dictionary" has the entry:
Fascinating (p. pr. & vb. n.) of Fascinate

I'm searching this input file for the term beginning with "fascinating", a space and an opening round bracket like so:
`rg.exe "^fascinating (?:[(])" dictionary -i --no-line-number --pcre2`

The result line is correct:
Fascinating (p. pr. & vb. n.) of Fascinate

However, the red colour match for this result line is:
**Fascinating (**
but I want to omit the opening round bracket for the red colouring because this is only the lookahead part.
The lookahead part should not be part of the match. Can this be done?

**Comment by BurntSushi** [comment, marked as answer]
Yes, if you use look-ahead. You aren't actually using look-ahead. I didn't test it, but try `^fascinating(?= [(])`.

See also: https://www.regular-expressions.info/lookaround.html

**Reply by datatraveller1** [reply]
Great, your solution works!
`rg.exe "^fascinating(?= [(])" dictionary -i --no-line-number --pcre2`

So,I have simply used look-ahead wrongly. Thank you so much!
